### PR TITLE
Add search button

### PR DESCRIPTION
**Pull request by gitdevmod**
Add search button
firefox:
image
ie11:
image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Comment by AnaelMobilia** [comment]
@gitdevmod You have a codestyle error 

FILE: glpi-project/glpi/css/styles.css
=> Line  2575 | ERROR | [x] Whitespace found at end of line

**Comment by gitdevmod** [comment]
in safari there is a space, I need to find a fix
capture d ecran 2017-11-28 a 07 51 16

**Comment by trasher** [comment]
I do not see what can cause the issue on safari; and I do not have it... Doesn't issue also appears on chrome(ium) as well? As far as I remember; safari uses webkit.

**Comment by gitdevmod** [comment]
With Chromium 62.0.3202.94 (ubuntu 17.10), I do not see the space.

**Comment by trasher** [comment]
OK... Cannot help you then, I do not have safari :/

**Comment by gitdevmod** [comment]
with ios11/safari, it's worse background is almost black. I tried:
`-webkit-appearance: none;
appearance: none;
`
with no luck
image

**Comment by trasher** [comment]
On hover with default theme; colors are not OK (bg color is the same than header bgcolor). I do not know if other themes are affected, each one must be tested.

Since the new element width is greater than original one, this may affect responsive view (I've not tested).

Note colors with default theme has changed:
screenshot-2017-11-29 glpi - changes - 3 1

screenshot-2017-11-29 glpi - changes - 3

Original style may be kept. Or not, I have no idea on this. @orthagh?

**Comment by orthagh** [comment]
will test now

**Comment by orthagh** [comment]
Or move the others icons on this second line

**Comment by gitdevmod** [comment]
I'm not expert in CSS maybe it's not the best way to align search bar
capture d ecran 2017-12-01 a 00 07 08
Still icon glass does not appear in ios/safari

**Comment by orthagh** [comment]
> I'm not expert in CSS maybe it's not the best way to align search bar
Seems ok for me

> Still icon glass does not appear in ios/safari
still a minor issue, i think we could skip this compatibility.
Safari has more and more css issues and have a very little market share.

Anyway, for the current pr, you have some coding standard issues 
-> https://travis-ci.org/glpi-project/glpi/jobs/309772577#L1229
Remove the traling space in css/styles.css

And don't forget @trasher comment:
> On hover with default theme; colors are not OK (bg color is the same than header bgcolor)

After fixing these 2 points, we'll merge i think

**Comment by gitdevmod** [comment]
@orthagh I do not see space at end of line 4299 and 4303 and I've not touched these lines.
@trasher Sorry I do not understand the problem with color style ?

File: css/styles.css [hunk 1]
```diff
@@ -4364,6 +4362,16 @@ input[type=checkbox].toggle_comments:checked ~ ul {
    #c_preference #debug_mode {
       display: none;
    }
+
```
Comment (orthagh) [inline css/styles.css]: @gitdevmod Here (4299) 3 trailings spaces

File: css/styles.css [hunk 2]
```diff
@@ -4364,6 +4362,16 @@ input[type=checkbox].toggle_comments:checked ~ ul {
    #c_preference #debug_mode {
       display: none;
    }
+   
+   #c_recherche form #champRecherche input{
+      width:100px;
+   }
+
```
Comment (orthagh) [inline css/styles.css]: And here (4303), also 3 trailing spaces

**Comment by orthagh** [comment]
Ex: 
image

**Comment by gitdevmod** [comment]
Weird lines does not match code image

**Comment by trasher** [comment]
@gitdevmod 
screenshot-2017-12-5 glpi - enclosures - 2

On hover:
screenshot-2017-12-5 glpi - enclosures - 2 1

**Comment by gitdevmod** [comment]
Thanks for merging. Mouse hover is not correct with all themes, I'll try this weeks to adjust css.

**Pull request merged**